### PR TITLE
Made generic

### PR DESCRIPTION
**Pull request by trong-le**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -37,11 +38,41 @@ test('proper configuration', t => {
 // 	}
 // });
 
+test('get taxi data', async t => {
+	const source = new FlightStatsSource();
+	try {
+		const taxi = await source.getTaxiData(config);
+		t.comment(JSON.stringify(taxi));
```
Comment (aloisbarreras) [inline test/test.js]: @trong-le get rid of all these t.comments

File: test/test.js [hunk 2]
```diff
@@ -37,11 +38,41 @@ test('proper configuration', t => {
 // 	}
 // });
 
+test('get taxi data', async t => {
+	const source = new FlightStatsSource();
+	try {
```
Comment (aloisbarreras) [inline test/test.js]: nothing needs to be in a try catch block

File: lib/index.js [hunk 3]
```diff
@@ -11,296 +11,474 @@ export default class FlightStatsSource extends Activity {
 
 	@singleS3StreamOutput('json')
 	async onTask(activityTask, config) {
-		let data = null, preTime = null;
+
+		let data = null;
 		switch (config.method) {
```
Comment (schnie) [inline lib/index.js]: switch statement could boil down to something like this...
`const data = this[config.method] ? await this[config.method](config) : {}`

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,5 @@
 node_modules
 .DS_Store
-test/config/**/*
 *.log
+*.config.js
+*.response.js
```
Comment (aloisbarreras) [inline .gitignore]: go ahead and get rid of the response.js line here

File: lib/index.js [hunk 5]
```diff
@@ -11,296 +11,411 @@ export default class FlightStatsSource extends Activity {
...
-			data = preData.fidsData;
-			break;
-			case 'taxi':
-			data = await this.getFlightStatusArrivalsByAirport(config);
-			break;
-			default:
-			break;
-		}
 
+		const data = this[config.method] ? await this[config.method](config) : {};
```
Comment (aloisbarreras) [inline lib/index.js]: I know Greg said to do it like this, but this is a much better way:
http://stackoverflow.com/a/912642

File: lib/index.js [hunk 6]
```diff
@@ -11,296 +11,411 @@ export default class FlightStatsSource extends Activity {
...
 
 		return data;
 	}
 
 	//MARK: FIDS Data
-	async fidsArrival(config) {
-		const uri = `${baseURI}fids/rest/v1/json/CVG/arrivals?appId=${config.appId}&appKey={config.appKey}&requestedFields=scheduledTime%2CscheduledDate%2Cflight%2CstatusCode%2Cremarks%2CactualTime%2CactualGateTime%2CcodesharesAsNames%2CairlineName%2CflightId&timeWindowBegin=720&timeWindowEnd=720&lateMinutes=15&useRunwayTimes=true&excludeCargoOnlyFlights=false`
+
+	async fidsArrival({ appId, appKey, requestedFields, timeWindowBegin, timeWindowEnd, lateMinutes }) {
+		const uri = `${baseURI}fids/rest/v1/json/CVG/arrivals`
```
Comment (aloisbarreras) [inline lib/index.js]: magic strings are bad.
break out the 'fids/rest/v1/json/CVG/arrivals' into a const and join it with baseURI in another const.

**Comment by aloisbarreras** [comment]
add a test for your getUniqueTaxiData method

**Comment by aloisbarreras** [comment]
addTimestamp() isn't an async function. You don't need to await it.

File: test/test.js [hunk 7]
```diff
@@ -30,9 +32,21 @@ test('get taxi data', async t => {
...
+		delete uniqueTaxi[`${i}`].timestamp;
+	}
+	t.deepEqual(JSON.stringify(uniqueTaxi), JSON.stringify(taxiData));
+	t.end();
+});
+
 test('get fids arrivals', async t => {
 	const source = new FlightStatsSource();
-	const fidsArr = await source.fidsArrival(config);
+	const fidsArr = source.fidsArrival(config);
```
Comment (aloisbarreras) [inline test/test.js]: fidsArrival is async, you have to await it.

**Comment by aloisbarreras** [comment]
you have a lot of mistakes regarding async/await, read this to clarify things:
https://www.twilio.com/blog/2015/10/asyncawait-the-hero-javascript-deserved.html

**Pull request merged**